contracts/cw20-escrow/src/state.rs

```diff
@@ -8,7 +8,7 @@ use cosmwasm_storage::{bucket, bucket_read, prefixed_read, Bucket, ReadonlyBucke
 
 use cw20::Cw20Coin;
 
-pub(crate) type EscrowBalance = (Vec<Coin>, Vec<Cw20Coin>);
+pub(crate) type GenericBalance = (Vec<Coin>, Vec<Cw20Coin>);
```

Comment:
I also favor a struct here:

```rust
pub struct GenericBalance {
  pub native: Vec<Coin>,
  pub cw20: Vec<Cw20Coin>,
}
```

We end up with `escrow.balance.native`, which is much more clear than `escrow.balance.0`

Reply:
And as I commented above, I think a fair bit of the logic can be added as methods to this struct rather than external functions. We may well want to pull this struct out later for use in other types that store lots of coins, so adding coins to it removing coins from it, and converting to `Vec<CosmosMsg>` would be nice to have as methods. Along with various constructor helpers.

Reply:
Good ideas. Thanks.